File: ios/Classes/FlutterWechatPlugin.m
#import "FlutterWechatPlugin.h"
#import <WechatOpenSDK/WXApi.h>
#import <WechatOpenSDK/WXApiObject.h>

// Define a block type for the cached request
typedef void(^WeChatURLHandler)(void);

@interface FlutterWechatPlugin () <WXApiDelegate, FlutterStreamHandler>
@property (nonatomic, strong) FlutterResult pendingLoginResult;
@property (nonatomic, assign) BOOL isRegistered;
@property (nonatomic, copy) WeChatURLHandler cachedURLHandler;
@property (nonatomic, strong) FlutterEventSink eventSink;
@end

@implementation FlutterWechatPlugin

+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
    FlutterMethodChannel* channel = [FlutterMethodChannel
                                   methodChannelWithName:@"flutter_wechat_plugin"
                                         binaryMessenger:[registrar messenger]];
    FlutterWechatPlugin* instance = [[FlutterWechatPlugin alloc] init];
    instance.isRegistered = NO;
    sharedInstance = instance; // Set static instance
    [registrar addMethodCallDelegate:instance channel:channel];
    [registrar addApplicationDelegate:instance];
    
    // Register event channel for response events
    FlutterEventChannel* eventChannel = [FlutterEventChannel
                                        eventChannelWithName:@"flutter_wechat_plugin/response_event"
                                              binaryMessenger:[registrar messenger]];
    [eventChannel setStreamHandler:instance];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
    if ([@"getPlatformVersion" isEqualToString:call.method]) {
        result([@"iOS " stringByAppendingString:[[UIDevice currentDevice] systemVersion]]);
    } else if ([@"registerApp" isEqualToString:call.method]) {
        [self registerApp:call result:result];
    } else if ([@"isWechatInstalled" isEqualToString:call.method]) {
        BOOL isInstalled = [WXApi isWXAppInstalled];
        result(@(isInstalled));
    } else if ([@"wechatLogin" isEqualToString:call.method]) {
        [self wechatLogin:result];
    } else if ([@"shareText" isEqualToString:call.method]) {
        [self shareText:call result:result];
    } else if ([@"shareNetworkImage" isEqualToString:call.method]) {
        [self shareNetworkImage:call result:result];
    } else if ([@"shareNetworkImageToScene" isEqualToString:call.method]) {
        [self shareNetworkImageToScene:call result:result];
    } else if ([@"shareLink" isEqualToString:call.method]) {
        [self shareLink:call result:result];
    } else if ([@"openMiniProgram" isEqualToString:call.method]) {
        [self openMiniProgram:call result:result];
    } else {
        result(FlutterMethodNotImplemented);
    }
}

- (void)registerApp:(FlutterMethodCall *)call result:(FlutterResult)result {
    NSDictionary *args = call.arguments;
    NSString *appId = args[@"appId"];
    NSString *universalLink = args[@"universalLink"];

    if (!appId || !universalLink) {
        result([FlutterError errorWithCode:@"INVALID_ARGUMENT" message:@"App ID and Universal Link are required" details:nil]);
        return;
    }

    BOOL success = [WXApi registerApp:appId universalLink:universalLink];
    self.isRegistered = success;

    if (success) {
        if (self.cachedURLHandler) {
            self.cachedURLHandler();
            self.cachedURLHandler = nil;
        }
    }
    result(@(success));
}

- (void)wechatLogin:(FlutterResult)result {
    if (!self.isRegistered) {
        result([FlutterError errorWithCode:@"WECHAT_NOT_REGISTERED" 
                                   message:@"WeChat is not registered. Call registerApp first." 
                                   details:nil]);
        return;
    }
    if (![WXApi isWXAppInstalled]) {
        result([FlutterError errorWithCode:@"WECHAT_NOT_INSTALLED" 
                                   message:@"WeChat is not installed" 
                                   details:nil]);
        return;
    }
    
    self.pendingLoginResult = result;
    
    SendAuthReq *req = [[SendAuthReq alloc] init];
    req.scope = @"snsapi_userinfo";
    req.state = @"flutter_wechat_plugin_state";
    
    [WXApi sendReq:req completion:^(BOOL success) {
        if (!success) {
            self.pendingLoginResult = nil;
            result([FlutterError errorWithCode:@"SEND_REQUEST_FAILED" 
                                       message:@"Failed to send login request" 
                                       details:nil]);
        }
    }];
}

#pragma mark - WXApiDelegate

- (void)onResp:(BaseResp *)resp {
    if ([resp isKindOfClass:[SendAuthResp class]]) {
        if (self.pendingLoginResult) {
            SendAuthResp *authResp = (SendAuthResp *)resp;
            NSMutableDictionary *response = [NSMutableDictionary dictionary];
            response[@"type"] = @"auth";
            response[@"errCode"] = @(authResp.errCode);
            if (authResp.code) response[@"code"] = authResp.code;
            if (authResp.state) response[@"state"] = authResp.state;
            if (authResp.lang) response[@"lang"] = authResp.lang;
            if (authResp.country) response[@"country"] = authResp.country;
            if (authResp.errStr) response[@"errStr"] = authResp.errStr;

            // This is the dual-callback mechanism from the Android implementation.
            // We resolve the future of the wechatLogin() call.
            self.pendingLoginResult(response);
            self.pendingLoginResult = nil;

            // We also send the event to the stream.
            [[self class] sendEventToDart:response];
        }
    } else if ([resp isKindOfClass:[SendMessageToWXResp class]]) {
        SendMessageToWXResp *messageResp = (SendMessageToWXResp *)resp;
        NSMutableDictionary *response = [NSMutableDictionary dictionary];
        response[@"type"] = @"share";
        response[@"errCode"] = @(messageResp.errCode);
        if (messageResp.errStr) response[@"errStr"] = messageResp.errStr;
        
        // Send the event to the stream
        [[self class] sendEventToDart:response];
    } else if ([resp isKindOfClass:[WXLaunchMiniProgramResp class]]) {
        // Handle mini program response
        WXLaunchMiniProgramResp *miniProgramResp = (WXLaunchMiniProgramResp *)resp;
        NSMutableDictionary *miniProgramResult = [@{
            @"errCode": @(miniProgramResp.errCode),
            @"errStr": miniProgramResp.errStr ?: @"",
            @"type": @"miniProgram"
        } mutableCopy];
        
        if (miniProgramResp.extMsg) {
            miniProgramResult[@"extMsg"] = miniProgramResp.extMsg;
        }
        
        [[self class] sendEventToDart:miniProgramResult];
    }
}

// Static instance to access from class methods
static FlutterWechatPlugin* sharedInstance = nil;

// Helper to send events to the Dart stream
+ (void)sendEventToDart:(NSDictionary *)event {
    if (sharedInstance && sharedInstance.eventSink) {
        sharedInstance.eventSink(event);
    }
}


#pragma mark - Application Delegate Methods

- (BOOL)application:(UIApplication *)application openURL:(NSURL *)url options:(NSDictionary<UIApplicationOpenURLOptionsKey, id> *)options {
    if (self.isRegistered) {
        return [WXApi handleOpenURL:url delegate:self];
    } else {
        self.cachedURLHandler = ^{
            [WXApi handleOpenURL:url delegate:self];
        };
        return NO;
    }
}

- (BOOL)application:(UIApplication *)application continueUserActivity:(NSUserActivity *)userActivity restorationHandler:(void (^)(NSArray * _Nullable))restorationHandler {
    if (self.isRegistered) {
        return [WXApi handleOpenUniversalLink:userActivity delegate:self];
    } else {
        self.cachedURLHandler = ^{
            [WXApi handleOpenUniversalLink:userActivity delegate:self];
        };
        return NO;
    }
}

#pragma mark - FlutterStreamHandler

- (FlutterError*)onListenWithArguments:(id)arguments eventSink:(FlutterEventSink)eventSink {
    self.eventSink = eventSink;
    return nil;
}

- (FlutterError*)onCancelWithArguments:(id)arguments {
    self.eventSink = nil;
    return nil;
}

#pragma mark - Share Methods

- (void)shareText:(FlutterMethodCall *)call result:(FlutterResult)result {
    if (!self.isRegistered) {
        result([FlutterError errorWithCode:@"WECHAT_NOT_REGISTERED" 
                                   message:@"WeChat is not registered. Call registerApp first." 
                                   details:nil]);
        return;
    }
    
    NSString *text = call.arguments[@"text"];
    if (!text) {
        result([FlutterError errorWithCode:@"INVALID_ARGUMENT" 
                                   message:@"Text is required" 
                                   details:nil]);
        return;
    }
    
    SendMessageToWXReq *req = [[SendMessageToWXReq alloc] init];
    req.scene = WXSceneSession; // Default to session (friends)
    req.bText = YES;
    req.text = text;
    
    [WXApi sendReq:req completion:^(BOOL success) {
        result(@(success));
    }];
}

- (void)shareNetworkImage:(FlutterMethodCall *)call result:(FlutterResult)result {
    if (!self.isRegistered) {
        result([FlutterError errorWithCode:@"WECHAT_NOT_REGISTERED" 
                                   message:@"WeChat is not registered. Call registerApp first." 
                                   details:nil]);
        return;
    }
    
    NSString *imageUrl = call.arguments[@"imageUrl"];
    if (!imageUrl) {
        result([FlutterError errorWithCode:@"INVALID_ARGUMENT" 
                                   message:@"Image URL is required" 
                                   details:nil]);
        return;
    }
    
    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
        NSData *imageData = [NSData dataWithContentsOfURL:[NSURL URLWithString:imageUrl]];
        
        dispatch_async(dispatch_get_main_queue(), ^{
            if (!imageData) {
                result([FlutterError errorWithCode:@"DOWNLOAD_FAILED" 
                                           message:@"Failed to download image" 
                                           details:nil]);
                return;
            }
            
            WXImageObject *imageObject = [WXImageObject object];
            imageObject.imageData = imageData;
            
            WXMediaMessage *message = [WXMediaMessage message];
            message.mediaObject = imageObject;
            
            // Create thumbnail
            UIImage *image = [UIImage imageWithData:imageData];
            if (image) {
                CGSize thumbnailSize = CGSizeMake(100, 100);
                UIGraphicsBeginImageContextWithOptions(thumbnailSize, NO, 0.0);
                [image drawInRect:CGRectMake(0, 0, thumbnailSize.width, thumbnailSize.height)];
                UIImage *thumbnailImage = UIGraphicsGetImageFromCurrentImageContext();
                UIGraphicsEndImageContext();
                
                if (thumbnailImage) {
                    message.thumbData = UIImageJPEGRepresentation(thumbnailImage, 0.8);
                }
            }
            
            SendMessageToWXReq *req = [[SendMessageToWXReq alloc] init];
            req.scene = WXSceneSession; // Default to session (friends)
            req.message = message;
            req.bText = NO;
            
            [WXApi sendReq:req completion:^(BOOL success) {
                result(@(success));
            }];
        });
    });
}

- (void)shareNetworkImageToScene:(FlutterMethodCall *)call result:(FlutterResult)result {
    if (!self.isRegistered) {
        result([FlutterError errorWithCode:@"WECHAT_NOT_REGISTERED" 
                                   message:@"WeChat is not registered. Call registerApp first." 
                                   details:nil]);
        return;
    }
    
    NSString *imageUrl = call.arguments[@"imageUrl"];
    NSNumber *sceneNumber = call.arguments[@"scene"];
    
    if (!imageUrl) {
        result([FlutterError errorWithCode:@"INVALID_ARGUMENT" 
                                   message:@"Image URL is required" 
                                   details:nil]);
        return;
    }
    
    enum WXScene scene = WXSceneSession; // Default to session
    if (sceneNumber) {
        int sceneInt = [sceneNumber intValue];
        if (sceneInt == 1) {
            scene = WXSceneTimeline; // Timeline (moments)
        } else if (sceneInt == 2) {
            scene = WXSceneFavorite; // Favorites
        }
    }
    
    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
        NSData *imageData = [NSData dataWithContentsOfURL:[NSURL URLWithString:imageUrl]];
        
        dispatch_async(dispatch_get_main_queue(), ^{
            if (!imageData) {
                result([FlutterError errorWithCode:@"DOWNLOAD_FAILED" 
                                           message:@"Failed to download image" 
                                           details:nil]);
                return;
            }
            
            WXImageObject *imageObject = [WXImageObject object];
            imageObject.imageData = imageData;
            
            WXMediaMessage *message = [WXMediaMessage message];
            message.mediaObject = imageObject;
            
            // Create thumbnail
            UIImage *image = [UIImage imageWithData:imageData];
            if (image) {
                CGSize thumbnailSize = CGSizeMake(100, 100);
                UIGraphicsBeginImageContextWithOptions(thumbnailSize, NO, 0.0);
                [image drawInRect:CGRectMake(0, 0, thumbnailSize.width, thumbnailSize.height)];
                UIImage *thumbnailImage = UIGraphicsGetImageFromCurrentImageContext();
                UIGraphicsEndImageContext();
                
                if (thumbnailImage) {
                    message.thumbData = UIImageJPEGRepresentation(thumbnailImage, 0.8);
                }
            }
            
            SendMessageToWXReq *req = [[SendMessageToWXReq alloc] init];
            req.scene = scene;
            req.message = message;
            req.bText = NO;
            
            [WXApi sendReq:req completion:^(BOOL success) {
                result(@(success));
            }];
        });
    });
}

- (void)shareLink:(FlutterMethodCall *)call result:(FlutterResult)result {
    if (!self.isRegistered) {
        result([FlutterError errorWithCode:@"WECHAT_NOT_REGISTERED" 
                                   message:@"WeChat is not registered. Call registerApp first." 
                                   details:nil]);
        return;
    }
    
    NSString *url = call.arguments[@"url"];
    NSString *title = call.arguments[@"title"];
    NSString *description = call.arguments[@"description"];
    
    if (!url || !title) {
        result([FlutterError errorWithCode:@"INVALID_ARGUMENT" 
                                   message:@"URL and title are required" 
                                   details:nil]);
        return;
    }
    
    WXWebpageObject *webpageObject = [WXWebpageObject object];
    webpageObject.webpageUrl = url;
    
    WXMediaMessage *message = [WXMediaMessage message];
    message.title = title;
    message.description = description ?: @"";
    message.mediaObject = webpageObject;
    
    SendMessageToWXReq *req = [[SendMessageToWXReq alloc] init];
    req.scene = WXSceneSession; // Default to session (friends)
    req.message = message;
    req.bText = NO;
    
    [WXApi sendReq:req completion:^(BOOL success) {
        result(@(success));
    }];
}

- (void)openMiniProgram:(FlutterMethodCall *)call result:(FlutterResult)result {
    if (!self.isRegistered) {
        result([FlutterError errorWithCode:@"WECHAT_NOT_REGISTERED" 
                                   message:@"WeChat is not registered. Call registerApp first." 
                                   details:nil]);
        return;
    }
    
    NSString *username = call.arguments[@"username"];
    NSString *path = call.arguments[@"path"];
    NSNumber *miniProgramTypeNumber = call.arguments[@"miniProgramType"];
    
    if (!username) {
        result([FlutterError errorWithCode:@"INVALID_ARGUMENT" 
                                   message:@"Username is required" 
                                   details:nil]);
        return;
    }
    
    // Convert miniProgramType from int to WXMiniProgramType
    WXMiniProgramType miniProgramType = WXMiniProgramTypeRelease; // Default
    if (miniProgramTypeNumber) {
        int typeInt = [miniProgramTypeNumber intValue];
        if (typeInt == 1) {
            miniProgramType = WXMiniProgramTypeTest;
        } else if (typeInt == 2) {
            miniProgramType = WXMiniProgramTypePreview;
        }
    }
    
    WXLaunchMiniProgramReq *launchMiniProgramReq = [WXLaunchMiniProgramReq object];
    launchMiniProgramReq.userName = username;
    launchMiniProgramReq.path = (path && ![path isEqual:[NSNull null]]) ? path : nil;
    launchMiniProgramReq.miniProgramType = miniProgramType;
    
    [WXApi sendReq:launchMiniProgramReq completion:^(BOOL success) {
        result(@(success));
    }];
}

@end

File: ios/Classes/FlutterWechatPlugin.swift
import Flutter
import UIKit
import WechatOpenSDK

public class FlutterWechatPlugin: NSObject, FlutterPlugin, WXApiDelegate {
  private var pendingResult: FlutterResult?
  
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "flutter_wechat_plugin", binaryMessenger: registrar.messenger())
    let instance = FlutterWechatPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("iOS " + UIDevice.current.systemVersion)
    case "registerApp":
      if let args = call.arguments as? [String: Any],
         let appId = args["appId"] as? String {
        registerApp(appId: appId, result: result)
      } else {
        result(FlutterError(code: "INVALID_ARGUMENT", message: "App ID is required", details: nil))
      }
    case "isWechatInstalled":
      let isInstalled = WXApi.isWXAppInstalled()
      print("FlutterWechatPlugin: WeChat installed check: \(isInstalled)")
      result(isInstalled)
    case "wechatLogin":
      wechatLogin(result: result)
    case "shareText":
      if let args = call.arguments as? [String: Any],
         let text = args["text"] as? String {
        shareText(text: text, result: result)
      } else {
        result(FlutterError(code: "INVALID_ARGUMENT", message: "Text is required", details: nil))
      }
    case "shareImage":
      if let args = call.arguments as? [String: Any],
         let imagePath = args["imagePath"] as? String {
        shareImage(imagePath: imagePath, result: result)
      } else {
        result(FlutterError(code: "INVALID_ARGUMENT", message: "Image path is required", details: nil))
      }
    case "shareLink":
      if let args = call.arguments as? [String: Any],
         let url = args["url"] as? String,
         let title = args["title"] as? String {
        let description = args["description"] as? String
        let thumbnailPath = args["thumbnailPath"] as? String
        shareLink(url: url, title: title, description: description, thumbnailPath: thumbnailPath, result: result)
      } else {
        result(FlutterError(code: "INVALID_ARGUMENT", message: "URL and title are required", details: nil))
      }
    case "shareNetworkImage":
      if let args = call.arguments as? [String: Any],
         let imageUrl = args["imageUrl"] as? String {
        shareNetworkImage(imageUrl: imageUrl, result: result)
      } else {
        result(FlutterError(code: "INVALID_ARGUMENT", message: "Image URL is required", details: nil))
      }
    default:
      result(FlutterMethodNotImplemented)
    }
  }
  
  private func registerApp(appId: String, result: @escaping FlutterResult) {
    print("FlutterWechatPlugin: Registering WeChat app with ID: \(appId)")
    let success = WXApi.registerApp(appId, universalLink: "")
    print("FlutterWechatPlugin: WeChat registration result: \(success)")
    result(success)
  }
  
  private func wechatLogin(result: @escaping FlutterResult) {
    if !WXApi.isWXAppInstalled() {
      result(FlutterError(code: "WECHAT_NOT_INSTALLED", message: "WeChat is not installed", details: nil))
      return
    }
    
    pendingResult = result
    
    let req = SendAuthReq()
    req.scope = "snsapi_userinfo"
    req.state = "wechat_sdk_demo_test"
    
    WXApi.send(req) { success in
      if !success {
        self.pendingResult?(FlutterError(code: "LOGIN_FAILED", message: "Failed to send login request", details: nil))
        self.pendingResult = nil
      }
    }
  }
  
  private func shareText(text: String, result: @escaping FlutterResult) {
    if !WXApi.isWXAppInstalled() {
      result(FlutterError(code: "WECHAT_NOT_INSTALLED", message: "WeChat is not installed", details: nil))
      return
    }
    
    let message = WXMediaMessage()
    message.title = text
    message.description = text
    message.mediaObject = WXTextObject()
    (message.mediaObject as! WXTextObject).contentText = text
    
    let req = SendMessageToWXReq()
    req.bText = false
    req.message = message
    req.scene = Int32(WXSceneSession.rawValue)
    
    WXApi.send(req) { success in
      result(success)
    }
  }
  
  private func shareImage(imagePath: String, result: @escaping FlutterResult) {
    if !WXApi.isWXAppInstalled() {
      result(FlutterError(code: "WECHAT_NOT_INSTALLED", message: "WeChat is not installed", details: nil))
      return
    }
    
    guard let image = UIImage(contentsOfFile: imagePath) else {
      result(FlutterError(code: "INVALID_IMAGE", message: "Invalid image file", details: nil))
      return
    }
    
    let message = WXMediaMessage()
    message.title = "Image"
    message.description = "Shared from Flutter"
    message.mediaObject = WXImageObject()
    (message.mediaObject as! WXImageObject).imageData = image.jpegData(compressionQuality: 0.8)
    
    // Set thumbnail
    let thumbImage = image.resized(to: CGSize(width: 150, height: 150))
    message.thumbData = thumbImage?.jpegData(compressionQuality: 0.8)
    
    let req = SendMessageToWXReq()
    req.bText = false
    req.message = message
    req.scene = Int32(WXSceneSession.rawValue)
    
    WXApi.send(req) { success in
      result(success)
    }
  }
  
  private func shareLink(url: String, title: String, description: String?, thumbnailPath: String?, result: @escaping FlutterResult) {
    if !WXApi.isWXAppInstalled() {
      result(FlutterError(code: "WECHAT_NOT_INSTALLED", message: "WeChat is not installed", details: nil))
      return
    }
    
    let webpageObject = WXWebpageObject()
    webpageObject.webpageUrl = url
    
    let message = WXMediaMessage()
    message.title = title
    message.description = description ?? ""
    message.mediaObject = webpageObject
    
    // Set thumbnail if provided
    if let thumbnailPath = thumbnailPath,
       let thumbImage = UIImage(contentsOfFile: thumbnailPath) {
      let resizedThumb = thumbImage.resized(to: CGSize(width: 150, height: 150))
      message.thumbData = resizedThumb?.jpegData(compressionQuality: 0.8)
    }
    
    let req = SendMessageToWXReq()
    req.bText = false
    req.message = message
    req.scene = Int32(WXSceneSession.rawValue)
    
    WXApi.send(req) { success in
      result(success)
    }
  }
  
  // MARK: - WXApiDelegate
  public func onReq(_ req: BaseReq) {
    // Handle incoming requests from WeChat
  }
  
  public func onResp(_ resp: BaseResp) {
    if let authResp = resp as? SendAuthResp {
      handleAuthResponse(authResp)
    }
  }
  
  private func shareNetworkImage(imageUrl: String, result: @escaping FlutterResult) {
    if !WXApi.isWXAppInstalled() {
      result(FlutterError(code: "WECHAT_NOT_INSTALLED", message: "WeChat is not installed", details: nil))
      return
    }
    
    // 异步下载图片
    downloadImage(from: imageUrl) { [weak self] image in
      DispatchQueue.main.async {
        if let image = image {
          self?.shareDownloadedImage(image: image, result: result)
        } else {
          result(FlutterError(code: "DOWNLOAD_FAILED", message: "Failed to download image", details: nil))
        }
      }
    }
  }
  
  private func downloadImage(from urlString: String, completion: @escaping (UIImage?) -> Void) {
    guard let url = URL(string: urlString) else {
      completion(nil)
      return
    }
    
    URLSession.shared.dataTask(with: url) { data, response, error in
      if let error = error {
        print("FlutterWechatPlugin: Error downloading image: \(error.localizedDescription)")
        completion(nil)
        return
      }
      
      guard let data = data, let image = UIImage(data: data) else {
        completion(nil)
        return
      }
      
      completion(image)
    }.resume()
  }
  
  private func shareDownloadedImage(image: UIImage, result: @escaping FlutterResult) {
    let message = WXMediaMessage()
    message.title = "Network Image"
    message.description = "Shared from Flutter"
    message.mediaObject = WXImageObject()
    (message.mediaObject as! WXImageObject).imageData = image.jpegData(compressionQuality: 0.8)
    
    // Set thumbnail
    let thumbImage = image.resized(to: CGSize(width: 150, height: 150))
    message.thumbData = thumbImage?.jpegData(compressionQuality: 0.8)
    
    let req = SendMessageToWXReq()
    req.bText = false
    req.message = message
    req.scene = Int32(WXSceneSession.rawValue)
    
    WXApi.send(req) { success in
      result(success)
    }
  }
  
  private func handleAuthResponse(_ resp: SendAuthResp) {
    guard let result = pendingResult else { return }
    pendingResult = nil
    
    if resp.errCode == WXSuccess.rawValue {
      let authData: [String: Any] = [
        "code": resp.code ?? "",
        "state": resp.state ?? "",
        "lang": resp.lang ?? "",
        "country": resp.country ?? ""
      ]
      result(authData)
    } else {
      let errorMessage = resp.errStr ?? "Login failed"
      result(FlutterError(code: "LOGIN_FAILED", message: errorMessage, details: nil))
    }
  }
}

// MARK: - UIImage Extension
extension UIImage {
  func resized(to size: CGSize) -> UIImage? {
    UIGraphicsBeginImageContextWithOptions(size, false, scale)
    defer { UIGraphicsEndImageContext() }
    draw(in: CGRect(origin: .zero, size: size))
    return UIGraphicsGetImageFromCurrentImageContext()
  }
}
